test/test.dart

```diff
@@ -234,27 +240,43 @@ Future<void> tryDeleteFile(FileSystemEntity fileEntity, {bool recursive = false}
   }
 }
 
-void parseTestNameFromArguments(List<String>? arguments) {
-  if (testName != null || arguments == null || arguments.isEmpty) {
-    return;
-  }
+Map<String, String?> testArguments(List<String>? arguments) {
+  Map<String, String?> testArgs = {};
+    final parser = ArgParser()
+      ..addOption("name")
+      ..addOption(argBaasUrl)
+      ..addOption(argBaasCluster)
+      ..addOption(argBaasApiKey)
+      ..addOption(argBaasPrivateApiKey)
+      ..addOption(argBaasProjectId);
+
+  final result = parser.parse(arguments ?? []);
+    testArgs
+      ..addArgument(result, "name")
+      ..addArgument(result, argBaasUrl)
+      ..addArgument(result, argBaasCluster)
+      ..addArgument(result, argBaasApiKey)
+      ..addArgument(result, argBaasPrivateApiKey)
+    ..addArgument(result, argBaasProjectId);
+  return testArgs;
+}
 
-  int nameArgIndex = arguments.indexOf("--name");
-  if (nameArgIndex >= 0 && arguments.length > nameArgIndex) {
-    testName = arguments[nameArgIndex + 1];
+extension on Map<String, String?> {
```

Comment:
What's the benefit of this extension. Why not just use regular method.
```
_addArgument(testArgs, result, "name");
_addArgument(testArgs, result, argBaasUrl);
```
as it is it is exposed in all test files since all import `test.dart`